pkg/compactor/compactor.go

```diff
@@ -165,22 +193,37 @@ type Compactor struct {
 
 // NewCompactor makes a new Compactor.
 func NewCompactor(compactorCfg Config, storageCfg cortex_tsdb.BlocksStorageConfig, logger log.Logger, registerer prometheus.Registerer) (*Compactor, error) {
-	createDependencies := func(ctx context.Context) (objstore.Bucket, tsdb.Compactor, compact.Planner, error) {
-		bucketClient, err := bucket.NewClient(ctx, storageCfg.Bucket, "compactor", logger, registerer)
-		if err != nil {
-			return nil, nil, nil, errors.Wrap(err, "failed to create the bucket client")
+	createBucketClient := func(ctx context.Context) (objstore.Bucket, error) {
```

Comment:
Decoupled from `createBlocksCompactor()` because I prefer to not let downstream projects customise it. The bucket is created based on the provided `storageCfg` and it should continue to be.